meson.build

```diff
@@ -990,11 +990,22 @@ if want_bpf_framework == 'false'
 else
         # Support 'versioned' clang/llvm-strip binaries, as seen on Debian/Ubuntu
         # (like clang-10/llvm-strip-10)
-        clang_bin = cc.get_id() == 'clang' ? cc.cmd_array()[0] : 'clang'
-        if meson.is_cross_build() or clang_bin.contains('afl-clang') or clang_bin.contains('hfuzz-clang')
-                clang_bin = 'clang'
+        if meson.is_cross_build() or cc.get_id() != 'clang' or cc.cmd_array()[0].contains('afl-clang') or cc.cmd_array()[0].contains('hfuzz-clang')
+                clang = find_program('clang', required : bpf_framework_required)
+                # Assume that the required flags are supported.
+                clang_supports_flags = true
+        else
+                clang = find_program(cc.cmd_array()[0], required : bpf_framework_required)
```

Comment:
What should this do when cc.cmd_array() is `['ccache', 'clang']`?

Reply:
@eli-schwartz it shouldn't rely on `cc.cmd_array()[0]` I think :-) That's a good point. Thanks!

Reply:
Quite so. ;)

Honestly, you can just use the entire cmd_array() directly as the found compiler, if you need to use it directly anywhere as a custom_target or run_target command. A found program and a cmd_array() are going to be interchangeable.

And setting `required: bpf_framework_required` (a bool value) is pointless, since the `meson.get_compiler('c')` found compiler will always exist, and can never not be found. So, you can set `clang_found = clang.found()` in the if, and unconditionally set `clang_found = true` in the else... and key off of that everywhere.